From c33f057b6eb031d19f1e6f39c817d6cb1ec61cbf Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Mon, 1 Sep 2025 14:53:16 +0200
Subject: [PATCH] Revert "feat(stm32mp2): remove PM in serial boot"

This reverts commit 5a51497b22a01f6c44ba46b9ac3bb1b97b51f3d2.

The kernel requires stm32mp2_pm even when booting from USB.
---
 plat/st/stm32mp2/bl31_plat_setup.c | 29 -----------------------------
 plat/st/stm32mp2/platform.mk       | 12 +++---------
 2 files changed, 3 insertions(+), 38 deletions(-)

diff --git a/plat/st/stm32mp2/bl31_plat_setup.c b/plat/st/stm32mp2/bl31_plat_setup.c
index ecf6ba6d9..c3bc60564 100644
--- a/plat/st/stm32mp2/bl31_plat_setup.c
+++ b/plat/st/stm32mp2/bl31_plat_setup.c
@@ -150,7 +150,6 @@ void bl31_plat_arch_setup(void)
 
 	stm32mp_gic_init();
 
-#if !STM32MP_UART_PROGRAMMER && !STM32MP_USB_PROGRAMMER
 	if (stm32mp_is_wakeup_from_standby()) {
 		/* Initialize the runtime services e.g. PSCI. */
 		runtime_svc_init();
@@ -164,7 +163,6 @@ void bl31_plat_arch_setup(void)
 		bl31_warm_entrypoint();
 		panic();
 	}
-#endif /* !STM32MP_UART_PROGRAMMER && !STM32MP_USB_PROGRAMMER */
 }
 
 void bl31_platform_setup(void)
@@ -211,30 +209,3 @@ entry_point_info_t *bl31_plat_get_next_image_ep_info(unsigned int type)
 
 	return next_image_info;
 }
-
-/* Only support system reset for serial boot and no low power feature to reduce BL31 size */
-#if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
-static void __dead2 stm32_system_reset(void)
-{
-	stm32mp_system_reset();
-}
-
-static const plat_psci_ops_t stm32_psci_ops = {
-	.system_reset = stm32_system_reset,
-};
-
-/* Stub PSCI platform functions */
-plat_local_state_t plat_get_target_pwr_state(unsigned int lvl,
-					     const plat_local_state_t *states,
-					     unsigned int ncpu)
-{
-	return 0U;
-}
-
-int plat_setup_psci_ops(uintptr_t sec_entrypoint,
-			const plat_psci_ops_t **psci_ops)
-{
-	*psci_ops = &stm32_psci_ops;
-	return 0;
-}
-#endif /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 542e382db..105a5496c 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -333,6 +333,7 @@ PLAT_BL_COMMON_SOURCES		+=	plat/st/stm32mp2/stm32mp2_otp_mirror.c
 
 PLAT_BL_COMMON_SOURCES		+=	drivers/st/iwdg/stm32_iwdg.c				\
 					drivers/st/reset/stm32mp2_reset.c			\
+					plat/st/stm32mp2/stm32mp2_context.c			\
 					plat/st/stm32mp2/stm32mp2_pwr.c				\
 					plat/st/stm32mp2/stm32mp2_syscfg.c
 
@@ -342,10 +343,6 @@ PLAT_BL_COMMON_SOURCES		+=	drivers/st/clk/clk-stm32-core.c				\
 
 PLAT_BL_COMMON_SOURCES  	+=	drivers/st/nvmem/stm32mp_tamp_nvram_mp2.c
 
-ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
-PLAT_BL_COMMON_SOURCES  	+=	plat/st/stm32mp2/stm32mp2_context.c
-endif
-
 BL2_SOURCES			+=	plat/st/stm32mp2/plat_bl2_mem_params_desc.c
 
 BL2_SOURCES			+=	drivers/st/crypto/stm32_hash.c				\
@@ -437,15 +434,12 @@ endif #STM32MP_M33_TDCID
 BL31_SOURCES			+=	${FDT_WRAPPERS_SOURCES}
 
 BL31_SOURCES			+=	plat/st/stm32mp2/bl31_plat_setup.c			\
-					plat/st/stm32mp2/stm32mp2_topology.c
-
-ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
-BL31_SOURCES			+=	plat/st/stm32mp2/stm32mp2_pm.c				\
+					plat/st/stm32mp2/stm32mp2_pm.c				\
+					plat/st/stm32mp2/stm32mp2_topology.c			\
 					plat/st/stm32mp2/stm32mp2_ca35ss.c			\
 					common/tf_crc32.c
 
 BL31_CPPFLAGS += -march=armv8-a+crc
-endif
 
 ifeq ($(STM32MP_M33_TDCID),0)
 BL31_SOURCES			+=	drivers/st/ddr/stm32mp2_ddr_helpers.c
